src/utils.c

```diff
@@ -931,3 +931,69 @@ ts_relation_size(Oid relid)
 
 	return ret;
 }
+
+#define STR_VALUE(str) #str
+#define NODE_CASE(name)                                                                            \
+	case T_##name:                                                                                 \
+		return STR_VALUE(name)
+
+/*
+ * Return a string with the name of the node.
+ *
+ */
+const char *
+ts_get_node_name(Path *path)
+{
+	switch (nodeTag(path))
+	{
+		case T_Path:
+			switch (path->pathtype)
+			{
+				NODE_CASE(SeqScan);
+				NODE_CASE(SampleScan);
+				NODE_CASE(SubqueryScan);
+				NODE_CASE(FunctionScan);
+				NODE_CASE(TableFuncScan);
+				NODE_CASE(ValuesScan);
+				NODE_CASE(CteScan);
+				NODE_CASE(WorkTableScan);
+				default:
+					return psprintf("Path (%d)", path->pathtype);
+			}
+			NODE_CASE(IndexPath);
+			NODE_CASE(BitmapHeapPath);
+			NODE_CASE(BitmapAndPath);
+			NODE_CASE(BitmapOrPath);
+			NODE_CASE(TidPath);
+			NODE_CASE(SubqueryScanPath);
+			NODE_CASE(ForeignPath);
```

Comment:
Shouldn't it be indented with the `case T_Path:`?

Reply:
fixed